libraries/Microsoft.Bot.Builder.Ai/SmartTalkMiddleware.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Ai
+{
+    public class SmartTalkMiddleware : IMiddleware
+    {
+        private readonly SmartTalk _smartTalk;
+        private readonly SmartTalkMiddlewareOptions _options;
+
+        public SmartTalkMiddleware(SmartTalkMiddlewareOptions options, HttpClient httpClient = null)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+
+            _smartTalk = new SmartTalk(options, httpClient);
```

Comment:
The lifecycle of Middleware in C# is long. In this case, if no HttpClient is passed in, I would create one here and pass it into your SmartTalk class (as a required argument).